src/charm.py

```diff
@@ -1182,8 +1182,9 @@ def _on_image_relation_joined(self, _: ops.RelationJoinedEvent) -> None:
             return
 
         clouds_yaml = state.charm_config.openstack_clouds_yaml
-        cloud = list(clouds_yaml["clouds"].keys())[0]
-        auth_map = clouds_yaml["clouds"][cloud]["auth"]
+        # unsubscriptable-object / thinks cloud_yaml is kind of a list
```

Comment:
Looking at charm_config it is `OpenStackCloudsYAML | None`, so it probably thinks it can be None, which it can be....